pov_tools/create_POV_data_files

```diff
@@ -0,0 +1,218 @@
+pngs=$(find . -type f -name '*.png')
+
+if [ "$files" -ge 2 ]; then
+	echo "OOPS! Too many .png image files found!"
+	echo ""
+	for strip in ${pngs[@]}; do
+		no_path="${strip##*/}"
+		echo ${no_path[*]}
+	done
+	echo ""
+	echo "There needs to be only ONE source .png image in the pov_tools folder."
+	echo ""
+	echo "Type 'r' - to remove ALL FILES with 'preview' in the name. "
+	echo "    If the additional filenames do not contain 'preview',"
+	echo "    manually delete all files except your source image."
+	echo ""
+	echo "Type q to quit."
+	read -p $'Press ENTER to start over.\n' fix
+	case $fix in
+		r) rm *preview*.png;
+			echo ""
+			echo "Files containing *preview* deleted if they existed"
+			read -n 1 -r -s -p $'Press any key to continue.\n'
+			exec "./create_POV_data_files";;
+		q) exit;;
+		*) exec "./create_POV_data_files";;
+	esac
+fi
+
+if [ "$files" -lt 1 ]; then
+	echo "OOPS! No .png image files found!"
+	echo ""
+	echo "Let's download an example Star Wars Logo for you to try :)"
+	echo "DOWNLOADING: 1024px-Star_Wars_Logo.svg.png .........."
+	echo ""
+	wget https://upload.wikimedia.org/wikipedia/commons/thumb/6/6c/Star_Wars_Logo.svg/1024px-Star_Wars_Logo.svg.png&>/dev/null &
+	logo=$"1024px-Star_Wars_Logo.svg.png"
+	have_logo=1
+	echo "Press q to quit, or"
+	read -n 1 -r -s -p $'press any other key to continue...\n' delay
+	if [ "$delay" = "q" ]; then exit; fi
+fi
+
+echo ""
+case $have_logo in
+	1) echo "Found the following source image file: $logo"
+		echo ""
+		sourcebase="${logo%.*}"
+		image="$logo";;
+	0)  source=$(find . -type f -name '*.png')
+		for strip in ${source[@]}; do
+			sourcename="${strip##*/}"
+			sourcebase="${sourcename%.*}"
+		done
+		echo "Found the following source image file: ${sourcename[*]}"
+		echo ""
+		# check for multiples and for filename contains 'preview'
+		recheck=$(ls 2>/dev/null -Ubad1 -- *.png | wc -l)
+		if [[ $sourcename == *"preview"* ]]; then
+			echo "OOPS! Source file name can not contain 'preview'."
+			echo "Please use a different file name."
+			echo "Press q to quit, or"
+			read -n 1 -r -s -p $'press any key to try again.\n' retry
+			if [ "$retry" = "q" ]; then exit; fi
+			exec "./create_POV_data_files"
+		elif [ $recheck -ge 2 ]; then
+			exec "./create_POV_data_files"
+		fi
+		image="$sourcename";;
+esac
+
+valid_option=0
+
+while [[ "$valid_option" != 1 ]]
+do
+    echo "Please select the settings you want to create the POV image with: "
+    echo "1 - Preset 1 (HEIGHT=144, LENGTH=39, OFFSET=18)"
+    echo "2 - Preset 2 (HEIGHT=97,  LENGTH=63, OFFSET=50)"
+    echo "3 - Preset 3 (HEIGHT=72,  LENGTH=50, OFFSET=15)"
+    echo "4 - Preset 4 (HEIGHT=72,  LENGTH=50, OFFSET=39)"
+    echo "5 - Height entry only"
+    echo "6 - Custom options"
+    echo "q - to quit"
+	read option
+	case $option in
+		"")	echo "! Some option is needed. Please try again"; echo "";;
+		1)	make IMAGE=$image OPTIONS="--height=144 --length=39 --offset=18"; HEIGHT=144 preset=1 valid_option=1;;
```

Comment:
I think it's weird to have several "make" invocations in this loop.
Why not have variables like "height", "width" and "length", have the loop adjust them, and once you exit the loop we run make IMAGE="$image" OPTIONS="--height=$height --length=$length offset=$offset"

Reply:
Just setting parameters now. One make command for all at the end.